Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,377 @@
+use PhpAmqpLib\Exchange\AMQPExchangeType;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver
+ * @package Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers
+ */
+class PhpAmqpLibDriver extends Service implements QueueDriverInterface
+{
+    use UsesSerializer;
+    use UsesExceptionHandlerTrait;
+
+    /**
+     * Maximum amount of time (in seconds) to wait for a message.
+     */
+    const WAIT_TIMEOUT = 10;
+
+    /**
+     * Maximum amount of time (in seconds) to set as an interval while consuming messages
+     */
+    const WAIT_PERIOD = 0.2;
+
+    /**
+     * Represents the amount of message to consume by iteration
+     */
+    const PREFETCH_COUNT = 1;
+
+    /*
+     * To use the default exchange pass an empty string
+     */
+    const EXCHANGE_NAME = '';
+
+    /**
+     * @var \AMQPQueue|null
+     */
+    private $queue;
+
+    /**
+     * @var int Time in ms
+     */
+    private $dequeueTimeMs = 0;
+
+    /**
+     * @var string
+     */
+    private $format = QueueDriverInterface::FORMAT_JSON;
+
+    /**
+     * @var AMQPChannel|null
+     */
+    private $channel;
+
+    /**
+     * @var \AMQPExchange|null
+     */
+    private $exchange;
+
+    /**
+     * @var string|null
+     */
+    private $host;
+
+    /**
+     * @var string|null
+     */
+    private $username;
+
+    /**
+     * @var string|null
+     */
+    private $password;
+
+    /**
+     * @var array|null
+     */
+    private $connections;
+
+    /**
+     * Method responsible to catch the parameters to configure the driver
+     *
+     * @param string $host
+     * @param string $username
+     * @param string $password
+     * @param string $format
+     */
+    public function configure($host, $username, $password, $format = QueueDriverInterface::FORMAT_JSON)
+    {
+        $this->host = $host;
+        $this->username = $username;
+        $this->password = $password;
+        $this->format = $format;
+    }
+
+    /**
+     * Method responsible to open the connection with the broker
+     *
+     * @param array $connections
+     * @param bool $shuffle - used to randomize the connections if exists more than one
+     * @throws \Exception
+     */
+    public function connect($connections = [], $shuffle = true)
+    {
+        try {
+            if ($shuffle) {
+                shuffle($connections);
+            }
+            $connection = AMQPStreamConnection::create_connection($connections, []);
+            $this->addConnection($connection);
```

Comment:
Can a specific process has more than 1 connection opened to rabbitMQ at the same time?

Reply:
By the driver it must to return the first successful connection. By this way, there is just one connection per process. However, this array connection can contains more than one connection, but working always with the first one.